packages/manager/apps/telecom/src/app/telecom/telephony/line/softphone/addDevice/add-device.controller.js

```diff
@@ -0,0 +1,10 @@
+export default class SoftphoneAddDeviceController {
+  /* @ngInject */
+  constructor() {
+    this.input = { name: null, recordLink: '-' };
+  }
+
+  checkInputAndGenerateLink() {
+    this.softphoneAddDeviceForm.$setSubmitted();
```

Comment:
```suggestion
```
I think you could base on creation of deviceId for to know if button can be available (if link was generated).

Reply:
this one is needed to generate the desired error message

Reply:
Are you sure because i don't see same an other place in code base. Maybe there is an other way for to have same that the other place

Reply:
Yes I am sure, this comes from https://stackoverflow.com/questions/42149236/angularjs-validation-how-do-i-trigger-validation-on-button-click